ui/article-renderer/src/main/java/org/cru/godtools/article/adapter/ArticleAdapter.java

```diff
@@ -0,0 +1,71 @@
+package org.cru.godtools.article.adapter;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.view.LayoutInflater;
+import android.view.ViewGroup;
+import org.cru.godtools.article.databinding.ListItemArticleBinding;
+import org.ccci.gto.android.common.recyclerview.adapter.SimpleDataBindingAdapter;
+import org.cru.godtools.articles.aem.model.Article;
+import org.cru.godtools.xml.model.Manifest;
+
+import java.util.List;
+
+public class ArticleAdapter extends SimpleDataBindingAdapter<ListItemArticleBinding> {
+    @Nullable
+    private List<Article> mArticles;
+    @Nullable
+    private Callback mCallback;
+    @Nullable
+    private Manifest mManifest;
+
+    @NonNull
+    @Override
+    protected ListItemArticleBinding onCreateViewDataBinding(@NonNull ViewGroup parent, int viewType) {
+        return ListItemArticleBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false);
+    }
+
+    @Override
+    protected void onBindViewDataBinding(ListItemArticleBinding binding, int position) {
+        assert mArticles != null : "Article must be defined to bind";
+        binding.setCallback(mCallback);
+        binding.setArticle(mArticles.get(position));
+        binding.setManifest(mManifest);
+    }
+
+    public void setCallbacks(Callback callbacks) {
+        this.mCallback = callbacks;
+        notifyItemRangeChanged(0, getItemCount());
+    }
+
+    public void setToolManifest(Manifest manifest) {
+        this.mManifest = manifest;
+        notifyItemRangeChanged(0, getItemCount());
+    }
+
+    @Override
+    protected void onViewDataBindingRecycled(@NonNull ListItemArticleBinding binding) {
+        super.onViewDataBindingRecycled(binding);
+        binding.setArticle(null);
+        binding.setCallback(null);
+    }
+
+    @Override
+    public int getItemCount() {
+        return mArticles != null ? mArticles.size() : 0;
+    }
+
+    public interface Callback {
+        void onArticleSelected(Article article);
+    }
+
+    /**
+     * This method is called to set the list of articles.  Used by Live Data to update list
+     *
+     * @param articles list of Articles
+     */
+    public void setArticles(List<Article> articles) {
+        mArticles = articles;
+        notifyDataSetChanged();
+    }
+}
```

Comment:
this functionally looks good, I would just re-organize the order of methods so that similar types of methods are together.

for example, put all the data setters together, keep all the lifecycle event methods (`on...`) together. put the `Callback` interface near the top of the class so it's easy to see for other developers when looking at the adapter.

when ordering the lifecycle event methods, try to order them in the sequence they will be called when possible, this allows other developers to quickly look through the lifecycle events to get an idea of the lifecycle of that component. In this component the sequence would be `create`, `bind`, `recycle`